.devcontainer/Dockerfile

```diff
@@ -1,18 +1,33 @@
-FROM mcr.microsoft.com/vscode/devcontainers/dotnet:1-8.0-jammy
+FROM debian:bookworm
 
-USER vscode
-WORKDIR /home/vscode
+# Insall curl for downloading required files
+RUN set -eux \
+    && apt-get update \
+    && apt-get install -y --no-install-recommends ca-certificates curl
+
+# Install prerequisites
+RUN set -eux \
+    && curl -o packages-microsoft-prod.deb https://packages.microsoft.com/config/debian/12/packages-microsoft-prod.deb \
+    && dpkg -i packages-microsoft-prod.deb \
+    && apt-get update \
+    && apt-get install -y libmsquic git libicu72 build-essential
+
+# Install .NET 8.0 and 9.0
+RUN set -eux \
+    && curl https://dot.net/v1/dotnet-install.sh -o dotnet-install.sh -L \
+    && chmod +x dotnet-install.sh \
+    && ./dotnet-install.sh -v latest -c 8.0 \
+    && ./dotnet-install.sh -v latest -q preview -c 9.0
+
+# Install Rust
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
 
-# Install necessary packages
+# Clenup
 RUN set -eux \
-    && wget -q https://packages.microsoft.com/config/ubuntu/22.04/packages-microsoft-prod.deb -O packages-microsoft-prod.deb \
-    && sudo dpkg -i packages-microsoft-prod.deb \
     && rm packages-microsoft-prod.deb \
-    && sudo apt update \
-    && sudo apt-get install -y libmsquic \
-    && sudo rm -rf /var/lib/apt/lists/* \
-    && sudo apt-get clean \
-    && curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
```

Comment:
It's best practice to do this any time you apt install something. It's not too important here since we don't ship this container so size doesn't really matter.

Reply:
Right the files would remain in the base layer. Maybe is just simpler to not call `apt-get clean` as this is a dev image and this extra size isn't that important.

Reply:
Sure, I'm fine to remove it for this dev container.